src/ripple/core/impl/TerminateHandler.cpp

```diff
@@ -18,61 +18,54 @@
 //==============================================================================
 
 #include <BeastConfig.h>
-#include <ripple/core/ThreadEntry.h>
+#include <ripple/core/TerminateHandler.h>
 #include <ripple/basics/Log.h>
+#include <ripple/beast/core/CurrentThreadName.h>
 
 #include <boost/coroutine/exceptions.hpp>
-#include <boost/thread/tss.hpp>
 #include <exception>
 #include <iostream>
+#include <typeinfo>
 
 namespace ripple {
 
-#ifndef NO_LOG_UNHANDLED_EXCEPTIONS
-static boost::thread_specific_ptr<std::string> threadName;
-
-namespace detail {
-void setThreadName(std::string name)
-{
-    try
-    {
-        threadName.reset(new std::string{std::move(name)});
-    }
-    catch(...)
-    {
-    }
-}
-}
-
 void terminateHandler()
 {
     if (std::current_exception())
     {
-        std::string const name = threadName.get() ? *threadName.get() : "Unknown";
+        std::string thread {beast::getCurrentThreadName()};
```

Comment:
If we change `getCurrentThreadName` to return a `boost::optional`, we can change this to:
`auto const thread = beast::getCurrentThreadName().value_or("Unknown");`